src/aggregation/metric/top_hits.rs

```diff
@@ -344,7 +366,7 @@ impl PartialEq for ComparableDocFeature {
 impl Eq for ComparableDocFeature {}
 
 #[derive(Clone, Serialize, Deserialize, Debug)]
-struct ComparableDocFeatures(Vec<ComparableDocFeature>, FieldRetrivalResult);
+struct ComparableDocFeatures(Vec<ComparableDocFeature>, IntermediateFieldRetrivalResult);
 
 impl Ord for ComparableDocFeatures {
     fn cmp(&self, other: &Self) -> std::cmp::Ordering {
```

Comment:
My understanding is that the `TopHitsCollector` needlessly access retrieved fields for every single document it collects. (Not even those who enter the Top K)

This is quite terrible. Can we at least open a ticket to fix this?

Reply:
Yes, that's quite terrible, I opened an issue here:
https://github.com/quickwit-oss/tantivy/issues/2347

Reply:
I fixed it in this PR, fields will be retrieved after segment collection now

Reply:
thx